albumentations/augmentations/transforms.py

```diff
@@ -1742,33 +1707,29 @@ def __init__(
     def apply(self, img: np.ndarray, gauss: np.ndarray, **params: Any) -> np.ndarray:
         return fmain.add_noise(img, gauss)
 
-    def get_params_dependent_on_targets(self, params: dict[str, Any]) -> dict[str, float]:
-        image = params["image"]
+    def get_params_dependent_on_data(self, params: dict[str, Any], data: dict[str, Any]) -> dict[str, float]:
+        shape = params["shape"]
         var = random.uniform(self.var_limit[0], self.var_limit[1])
         sigma = math.sqrt(var)
 
         if self.per_channel:
-            target_shape = image.shape
+            target_shape = shape
             if self.noise_scale_factor == 1:
                 gauss = random_utils.normal(self.mean, sigma, target_shape)
             else:
                 gauss = fmain.generate_approx_gaussian_noise(target_shape, self.mean, sigma, self.noise_scale_factor)
         else:
-            target_shape = image.shape[:2]
+            target_shape = shape[:2]
             if self.noise_scale_factor == 1:
                 gauss = random_utils.normal(self.mean, sigma, target_shape)
             else:
                 gauss = fmain.generate_approx_gaussian_noise(target_shape, self.mean, sigma, self.noise_scale_factor)
 
-            if image.ndim > MONO_CHANNEL_DIMENSIONS:
+            if shape[-1] > MONO_CHANNEL_DIMENSIONS:
```

Comment:
`shape[-1]` and `image.ndim` are quite different things.

And here should be `image.ndim`